dm/ctl/common/util.go

```diff
@@ -66,13 +66,80 @@ func PrintLines(format string, a ...interface{}) {
 
 // PrettyPrintResponse prints a PRC response prettily
 func PrettyPrintResponse(resp proto.Message) {
+	s, err := marshResponseToString(resp)
+	if err != nil {
+		PrintLines(errors.ErrorStack(err))
+	} else {
+		fmt.Println(s)
+	}
+}
+
+func marshResponseToString(resp proto.Message) (string, error) {
 	// encoding/json does not support proto Enum well
 	mar := jsonpb.Marshaler{EmitDefaults: true, Indent: "    "}
 	s, err := mar.MarshalToString(resp)
+	return s, errors.Trace(err)
+}
+
+// PrettyPrintResponseWithCheckTask prints a RPC response may contain response Msg with check-task's response prettily.
+// check-task's response may contain json-string when checking fail in `detail` field.
+// ugly code, but it is a little hard to refine this because needing to convert type.
+func PrettyPrintResponseWithCheckTask(resp proto.Message, subStr string) bool {
+	var (
+		err          error
+		found        bool
+		replacedStr  string
+		marshaledStr string
+		placeholder  = "PLACEHOLDER"
+	)
+	switch chr := resp.(type) {
+	case *pb.StartTaskResponse:
```

Comment:
we can use a unified `case *pb.StartTaskResponse, *pb.UpdateTaskResponse, *pb.CheckTaskResponse` as they have same logic code?